purchase_order_department_operating_unit/models/purchase_order.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+import logging
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    operating_unit_id = fields.Many2one(
+        related="department_id.operating_unit_id"
+    )
```

Comment:
คิดว่าควรใช้เป็น compute ดีกว่า เนื่องจากว่าถ้าใช้ @api.onchange จะส่งผลเฉพาะทำผ่าน UI เท่านั้น  ถ้า import ข้อมูลหรือสร้างข้อมูลผ่านทางอื่น จะไม่ได้คุณสมบัติตรงนี้

และแก้ไขค่า default ของ OU ให้สอดคล้องกับ department_id

```suggestion

    def _default_operating_unit_id(self):
        department_id = self.env.user.employee_id.department_id
        if department_id and department_id.operating_unit_id:
            return department_id.operating_unit_id.id
        return self.env["res.users"].operating_unit_default_get()

    operating_unit_id = fields.Many2one(
        compute="_compute_operating_unit_id",
        default=lambda self: self._default_operating_unit_id(),
        store=True,
    )

    @api.depends("department_id")
    def _compute_operating_unit_id(self):
        for rec in self:
            if rec.department_id and rec.department_id.operating_unit_id:
                rec.operating_unit_id = rec.department_id.operating_unit_id.id
            else:
                rec.operating_unit_id = False

```